src/api/one/profiler/AsyncProfiler.java

```diff
@@ -57,17 +54,31 @@ public static synchronized AsyncProfiler getInstance(String libPath) {
     }
 
     private static File extractEmbeddedLib() {
-        String resourceName = "/" + getPlatformTag() + "/libasyncProfiler.so";
-        InputStream in = AsyncProfiler.class.getResourceAsStream(resourceName);
-        if (in == null) {
+        String libraryPath = System.getProperty("one.profiler.libraryPath");
+        URL url = AsyncProfiler.class.getResource(libraryPath == null
+            ? ("/" + getPlatformTag() + "/libasyncProfiler.so")
+            : libraryPath
+        );
+
+        if (url == null) {
+            if (libraryPath != null) {
+                throw new IllegalArgumentException("Invalid embedded library path: " + libraryPath);
+            }
             return null;
         }
 
+        if (!url.getPath().isEmpty()) {
+            return new File(url.getPath());
+        }
+
         try {
             String extractPath = System.getProperty("one.profiler.extractPath");
             File file = File.createTempFile("libasyncProfiler-", ".so",
                     extractPath == null || extractPath.isEmpty() ? null : new File(extractPath));
-            try (FileOutputStream out = new FileOutputStream(file)) {
+            file.deleteOnExit();
```

Comment:
`deleteOnExit` does not work in many cases: if `System.halt` is called, if the process is terminated by the OOM killer, if JVM crashes. It's better to not leave garbage; deleting temporary .so after immediately loading it is a common practice.